### DISCUSSION TITLE
Python compatibility longterm

### DISCUSSION THREAD

**Discussion by clemens-fricke** [tataratat/gustaf#145] [Polls]
We are currently working with, at least internally, the goal that we support all Python versions from `3.6` onwards. 

When I look at the officially supported versions from the Python organization version `3.7` just entered `end of life`. I have already had it that once or twice I bumped into issues where my preferred solution to a problem was incompatible with version `3.6`. This will most likely increase the longer we keep older versions supported. 

In addition with each release, we increase the testing load for all versions.  

For me coming from this, the question arises should we change our support policy so that we keep up with the Python organizations' support for its versions?

This does not mean that we will intentionally restrict access to the releases to these versions, but only that we do not test for it anymore and new features are allowed to be used.

This might also apply to the other packages under the Tataratat umbrella.

**Comment by clemens-fricke** [comment]
I do not see a reason to keep the support up for the old versions. But I would like to hear some actual use case-specific reasons to keep the support up. I am more than willing to change my mind if there are legitimate reasons.

**Reply by j042** [reply]
After offline discussions, we decided to drop EOL versions. Thanks for bringing this up

**Comment by j042** [comment]
may have to drop 3.8 too, as `numpy` stops building wheels for 3.8. Anyone using 3.8?
